File: src/indicators/utils/Csv.java
package indicators.utils;
import indicators.collection.*;

public class Csv {
	public static PriceStream get(String file) throws Exception {		
		TextFile f = new TextFile(file); 
		PriceStream s = new PriceStream();
		while (f.iterator().hasNext()) {
			String line = f.iterator().next(); 
			String[] dat = line.split(",");
			double high = Double.parseDouble(dat[3]); 
			double low = Double.parseDouble(dat[4]);
			double open = Double.parseDouble(dat[2]);
			double close = Double.parseDouble(dat[5]);
			s.add(open, high, low, close);
		}
		return s;
	}
}

File: src/indicators/Csv.scala
package indicators

object Csv {  
  def get(file:String,capacity:Int=1000):PriceStream = {
    var s = new PriceStream(capacity)
    var lines = scala.io.Source.fromFile(file).getLines
    for (l<-lines) {
      var dat = l.split(",")
      var open = java.lang.Double.parseDouble(dat(2))
      var high = java.lang.Double.parseDouble(dat(3))
      var low = java.lang.Double.parseDouble(dat(4))
      var close = java.lang.Double.parseDouble(dat(5))
      s.add(open,high,low,close) 
    }
    s
  }
}  
 
object Test extends Application {
  var s = ForexStream.create(Csv.get("/home/seth/Desktop/aud_nzd.csv"),"AUDNZD",60)  
  println(s.close()+" "+s.high()+" "+s.low()+" "+s.open())
  println("size is "+s.size) 
}
